crates/ruff/src/rules/ruff/rules/pairwise_over_zipped.rs

```diff
@@ -0,0 +1,117 @@
+use std::num::ParseIntError;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::source_code::Stylist;
+use rustpython_parser::ast::{Expr, ExprKind};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+use crate::Range;
+use ruff_python_ast::helpers::unparse_constant;
+
+#[violation]
+pub struct PairwiseOverZipped {}
+
+impl Violation for PairwiseOverZipped {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Prefer `itertools.pairwise()` to `zip()` if trying to fetch an iterable's successive overlapping pairs.")
+    }
+}
+
+#[derive(Debug)]
+struct SliceInfo {
+    arg_name: String,
+    slice_start: Result<i64, ParseIntError>,
+    slice_end: Result<i64, ParseIntError>,
+}
+
+impl SliceInfo {
+    pub fn new(
+        arg_name: String,
+        slice_start: Result<i64, ParseIntError>,
+        slice_end: Result<i64, ParseIntError>,
+    ) -> Self {
+        Self {
+            arg_name,
+            slice_start,
+            slice_end,
+        }
+    }
+}
+
+// Get arg name, lower bound, and upper bound for an expression, if it's a slice
+fn get_slice_info(expr: &Expr, stylist: &Stylist) -> Option<SliceInfo> {
+    let ExprKind::Subscript { value, slice, .. } = &expr.node else {
+        return None;
+    };
+
+    let ExprKind::Name { id: arg_id, .. } = &value.node else {
+        return None;
+    };
+
+    let mut lower_bound = Ok(0i64);
+    let mut upper_bound = Ok(0i64);
```

Comment:
We can’t default *these* unrecognized things to 0 either: `a[x:-1]` is different from `a[0:-1]`.

So is `a[0:-1:2]`, by the way.

Reply:
@andersk brought up a few good edge cases. Could you add those to the test suite?

Reply:
Yup, will do